### ISSUE TITLE
Allow for alternative authentication mechanisms for the Octokit object on construction.

### ISSUE THREAD

**Issue by josephperrott** [actions/toolkit#106]
Currently for the `@actions/github` this library extends `@octokit/rest`'s `Octokit` class and in the `super` call assumes that the authentication is a standard token. ([code link](https://github.com/actions/toolkit/blob/fc9958ec6a57d1b08e9a387c49971faecd0f0405/packages/github/src/github.ts#L18))

Other authentication methods should be valid here as well.

**Comment by damccorm** [comment]
Looking at your issue, it looks like your use case is if you don't want to authenticate at all, right? I think I see a use case there, not sure if I see a use case for alternative auth methods - does that seem right?

**Comment by josephperrott** [comment]
The use case I specifically am running into is needing to authentication with a JWT.

**Comment by ericsciple** [comment]
I'm not sure whether we need to implement this.

The toolkit class is a thin wrapper convenience function for the common case `GITHUB_TOKEN` or PAT.

I could see the need if an input accepted a PAT or JWT and a function provided a single abstraction to do the right thing.

Otherwise I would recommend using the Octokit class directly.

Thoughts?

**Comment by mies386** [comment]
#106
go_som

**Comment by mies386** [comment]
#285 
Pastebin_master_social_login
#292

**Comment by mies386** [comment]
#106/€Cash.paypal;bitcoin<€9999€>$cash_come_login_pal_pay;#social_network
#285 
#293 
<#$€$€cash_download;Apps>Pay:pal@internet^^==GitHub.patch/h.a.c.k.b;a;n;k-S;p;a;r<=#kasse#=><server$€&@!^#*

**Comment by eine** [comment]
IMHO, `@actions/github` should not duplicate or wrap any of the features provided by `@octokit/rest`, but extend them. Therefore, the authentication and all regular operations should be done with octokit. When a feature from `@actions/github` is required, it should accept an existing octokit object as an argument. Therefore, it should not be required for `@actions/github` to handle authentication at all. For example:

```js
const octokit = new Octokit(getInput('token', { required: false }))
const ghkit = new github.GitHub(octokit)
```

instead of https://github.com/eine/issue-runner/blob/258088daba8c0e57368d08b2afbc5795cf282490/ts/main.ts#L39-L45

**Comment by ericsciple** [comment]
fixed with #314 since the value of github.GitHub expanded to support [runner proxy settings](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/about-self-hosted-runners#using-a-proxy-server-with-self-hosted-runners)

now you can use a different auth mechanism, but still get the proxy settings.

**Comment by iamstarkov** [comment]
@ericsciple can you remind me how to use different auth mechanism? I've read linked PR and dont really understand how it was meant to be achieved
